include/deal2lkit/utilities.h

```diff
@@ -460,4 +460,27 @@ void smart_delete (SmartPointer<TYPE> &sp)
     }
 }
 
+/**
+ *  A simple class to shift a vector by a scalar.
+ *  This function is deprecated in deal but needed in many codes
+ */
+
+template <typename VEC>
+void vector_shift(VEC &in_vec, double a_scalar)
+{
+#ifdef DEAL_II_WITH_CXX11
+  for (auto i : in_vec.locally_owned_elements())
+    {
+      in_vec[i] += a_scalar;
+    }
+#else
+  for (unsigned int i = 0; i<in_vec.size(); ++i)
```

Comment:
This is not the alternative way to do the above... the alternative way would be

~~~C++
for(IndexSet::Iterator it=in_vec.().begin(); it != in_vec.end(); ++it)
    in_vec[*it] += a_scalar;
~~~

which is *identical* to the CXX_11 version. The auto is just a way to write more compactly the above loop which uses explicitly the iterators...